lib/judges/baza.rb

```diff
@@ -219,6 +210,66 @@ def headers
     }
   end
 
+  def request_options_for_push(data, meta)
+    { connecttimeout: @timeout, timeout: @timeout }
+      .then { |req_opts| put_headers(req_opts) }
+      .then { |req_opts| put_body(req_opts, data) }
+      .then { |req_opts| put_meta(req_opts, meta) }
+  end
+
+  def put_headers(req_opts)
```

Comment:
@tank-bohr the idea of a function/method in OOP (or any other programming paradigm) is to enable code reuse. If you use the code only once, you inline it. You create a method when you use the code twice or more times.

Reply:
It's debatable. I believe that the main goal of extracting a function is to increase the level of abstraction. The ability to reuse code is a pleasant consequence. For me, it's obvious, and I believe we have a consensus about that in the industry.

- [five lines rule from Sandi Metz](https://thoughtbot.com/blog/sandi-metz-rules-for-developers#five-lines-per-method)
- [here](https://youtu.be/DC-pQPq0acs?feature=shared&t=180) is a very nice example of extracting the code with explanations.